### PR TITLE
Add projects service

### PR DESCRIPTION
**Pull request by benpankow**
Addresses #267. Adds a `project` service available at /project/ that allows for basic CRUD operations for projects.

Endpoints are:
`GET /project/`: Returns a list of all projects
`POST /project/`: Creates a new project
`GET /project/PROJECTID/`: Returns information for the project with id `PROJECTID`
`GET /project/filter/?key=value`: Filters projects by given key-value pairs
`DELETE /project/PROJECTID/`: Deletes the project with id `PROJECTID`
`PUT /project/`: Updates a project with the given `id`

Also includes relevant docs/tests for each endpoint.

### CODE REVIEW

File: documentation/docs/reference/services/Project.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+=====
+
+GET /project/PROJECTID/
+---------------------
+
+Returns the project with the id of `PROJECTID`.
+
+Response format:
+```
+{
```
Comment (bstrauch24) [inline documentation/docs/reference/services/Project.md]: Need: project description
Change: location -> room number, i.e. Siebel 1214

File: documentation/docs/reference/services/Project.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	"id": "52fdfc072182654f163f5f0f9a621d72",
+	"name": "Example Project 10",
+	"mentors": ["Jane Doe", "John Smith"],
+	"location": {
+		"description": "Example Location",
+		"latitude": 40.1138,
+		"longitude": -88.2249
+	}
+	"tags": ["BACKEND", "FRONTEND"],
+	"code": "A1"
```
Comment (bstrauch24) [inline documentation/docs/reference/services/Project.md]: Make this numeric, "universally unique id"

Comment (bstrauch24) [reply documentation/docs/reference/services/Project.md]: call it "number"

File: services/project/controller/controller.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not get updated project."))
+		return
+	}
+
+	json.NewEncoder(w).Encode(updated_project)
+}
+
+/*
+	Endpoint to update a project
```
Comment (bstrauch24) [inline services/project/controller/controller.go]: Self-descriptive function, leave out comment

**Comment by PatrickKan** [comment]
Could you also make sure favoriting is added to the documentation? Like the payload that needs to be sent to add/remove favorites

**Pull request merged**